Refund reasons references (#1657)
saleor/saleor-docs: docs/developer/payments/refunds.mdx

  It also means that `totalChargedAmount` includes `amountChargePending`, and `totalRefundedAmount` includes `amountRefundPending`.
- Authorized amounts, including `amountAuthorized` and `amountAuthorizePending`, are also considered processed. These amounts will contribute to the overall processed totals and impact the `overchargedAmount`.
  :::

## Refund settings

:::note
Available from 3.22
:::

> While `reason` field is optional and remains a plain text field, the `reasonReference` can be configured:

Reviewer: IMO we should firstly add short explanation what is this refund setting, and why it's usefull.

- Require staff user to pass `reasonReference` for all refunds.
- Make `reasonReference` optional

To change the settings, you can either use the Dashboard or a mutation

```graphql
mutation UpdateRefundSettings($id: ID) {
    refundSettingsUpdate(input: {
      reasonReferenceType: $id
    }) {
        reasonReferenceType
    }
}

> ```

Reviewer: You should also attach the example variables and show the response of this mutations. Use `Tab` for it. You can see the examples here: https://github.com/saleor/saleor-docs/pull/1674

> - input `reasonReferenceType` - the ID of a Model Type. All models that are instances of this type will be available to reference in the reasons.

Reviewer: `the ID of a Model Type.`
- `Model Type` is `Page Type` in the `API`, so probably we should mentioned both until it changed, especially that you describe the mutation

  It also means that `totalChargedAmount` includes `amountChargePending`, and `totalRefundedAmount` includes `amountRefundPending`.
- Authorized amounts, including `amountAuthorized` and `amountAuthorizePending`, are also considered processed. These amounts will contribute to the overall processed totals and impact the `overchargedAmount`.
  :::

## Refunds configuration

:::note
Available from 3.22
:::

Staff users with `MANAGE_SETTINGS` permission can access additional refunds configuration available in Dashboard and via API.

Configuration is global and reflected on every channel.

> Currently, `RefundSettings` type controls a single behavior: whether the "reason reference" is required when the refund is created or updated.

Reviewer: Maybe use the variable that is available in inputs
```suggestion
Currently, `RefundSettings` type controls a single behavior: whether the `reasonReference` is required when the refund is created or updated.
```

> ### Setting refund reason type requirement

Reviewer: We are using Chicago style convention, so each word should be capitalised:
```suggestion
### Setting Refund Reason Type Requirement
```

You can decide you want a strict policy to require a reason reference for every refund. Such reference will be a relation to a `Page` type. To enable that, you need first to create a Page Type dedicated for such refunds. You can do that in the Dashboard or by `pageTypeCreate` mutation.

Once you have a `PageType` ID, you can set it in the `refundSettingsUpdate` mutation.

<Tabs>
    <TabItem value="Mutation">
    ```graphql
    mutation SetRefundTypeRequired($pageTypeId:ID!){
      refundSettingsUpdate(input:{
        refundReasonReferenceType:$pageTypeId
      }){
        errors{
          message
        }
      }
    }
    ```
    </TabItem>
    <TabItem value="Variables">
    ```json
    {
      "pageTypeId": "your-id"
    }
    
    ```
    </TabItem>
</Tabs>

Now, every time a *staff user* tries to create a refund via `transactionActionRequest` or `orderGrantRefundCreate`/`orderGrantRefundUpdate` mutations (or via Dashboard), they will be required to provide a reason reference. The existing "reason" field will still be available, but it will be optional.

> Note that permitted `App` will be able to omit this setting. For the User, the ValidationError will be raised if the reason reference is not provided. App can omit this field for more powerful workflows.

Reviewer 2: OR
```suggestion
Note that permitted `App` will be able to omit this setting. For the `User`, the ValidationError will be raised if the reason reference is not provided. App can omit this field for more powerful workflows.
```
(to treat User word as `User` type - align with `App` usage)

> Note that permitted `App` will be able to omit this setting. For the User, the ValidationError will be raised if the reason reference is not provided. App can omit this field for more powerful workflows.

Reviewer 2: I would consider placing it inside:
```
:::note
<>
:::
```
To keep this information more visible (as it says that the validation works differently for the App)
